crates/sncast/src/helpers/interactive.rs

```diff
@@ -0,0 +1,193 @@
+        .map(|local_path| {
+            format!(
+                "Yes, local default account ({}).",
+                to_tilde_path(&local_path)
+            )
+        })
+    {
+        options.push(option);
+    }
+
+    let selection = Select::with_theme(&ColorfulTheme::default())
+        .with_prompt("Do you want to make this account default?")
+        .items(&options)
+        .default(0)
+        .interact()
+        .context("Failed to display selection dialog")?;
+
+    match options[selection].as_str() {
+        selected if selected.starts_with("Yes, global default") => {
+            if let Ok(global_path) = get_global_config_path() {
+                edit_config(&global_path, "default", "account", account)?;
+            }
+        }
+        selected if selected.starts_with("Yes, local default") => {
+            if let Ok(current_path) = env::current_dir() {
+                let current_path_utf8 = Utf8PathBuf::from_path_buf(current_path).map_err(|_| {
+                    anyhow!("Failed to convert current directory path to Utf8PathBuf")
+                })?;
+
+                if let Ok(local_path) = search_config_upwards_relative_to(&current_path_utf8) {
+                    edit_config(&local_path, "default", "account", account)?;
+                }
+            }
+        }
+        _ => {}
+    }
+
+    Ok(())
+}
+
+fn edit_config(config_path: &Utf8PathBuf, profile: &str, key: &str, value: &str) -> Result<()> {
+    let file_content = fs::read_to_string(config_path)?;
+
+    let mut toml_doc = file_content
+        .parse::<DocumentMut>()
+        .context("Failed to parse TOML")?;
+    update_config(&mut toml_doc, profile, key, value);
+
+    fs::write(config_path, toml_doc.to_string())?;
+
+    Ok(())
+}
+
+fn update_config(toml_doc: &mut DocumentMut, profile: &str, key: &str, value: &str) {
+    if !toml_doc.contains_key("sncast") {
+        toml_doc["sncast"] = Item::Table(Table::new());
+    }
+
+    let sncast_table = toml_doc
+        .get_mut("sncast")
+        .and_then(|item| item.as_table_mut())
+        .expect("Failed to create or access 'sncast' table");
+
+    if !sncast_table.contains_key(profile) {
+        sncast_table[profile] = Item::Table(Table::new());
+    }
+
+    let profile_table = sncast_table
+        .get_mut(profile)
+        .and_then(|item| item.as_table_mut())
+        .expect("Failed to create or access profile table");
+
+    profile_table[key] = Value::from(value).into();
+}
+
+fn to_tilde_path(path: &Utf8PathBuf) -> String {
+    if cfg!(not(target_os = "windows")) {
+        if let Some(home_dir) = dirs::home_dir() {
+            if let Ok(canonical_path) = path.canonicalize() {
+                if let Ok(stripped_path) = canonical_path.strip_prefix(&home_dir) {
+                    return format!("~/{}", stripped_path.to_string_lossy());
+                }
+            }
+        }
+    }
+
+    path.to_string()
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Let's also add a test case where there is no `[sncast]` section at all so it has to be created for a given profile.

And another case where there's `[sncast.default]` but no `[sncast.testnet]`, which has to be created.